nibabel/streamlines/trk.py

```diff
@@ -423,8 +424,20 @@ def save(self, fileobj):
             i4_dtype = np.dtype("<i4")  # Always save in little-endian.
             f4_dtype = np.dtype("<f4")  # Always save in little-endian.
 
+            # Make sure streamlines are in rasmm then send them to voxmm.
+            tractogram = self.tractogram.to_world(lazy=True)
+            affine_to_trackvis = get_affine_rasmm_to_trackvis(header)
+            tractogram = tractogram.apply_affine(affine_to_trackvis, lazy=True)
+            # Assume looping over the streamlines can be done only once.
+            tractogram = iter(tractogram)
+
             try:
-                first_item = next(iter(self.tractogram))
+                # Use the first element to check
```

Comment:
I guess you could refactor this out into:

```
def peek_next(iterable):
    next_item = next(iterable)
    return next_item, itertools.chain([next_item], iterable)
```

Reply:
Good idea. Will do.

Reply:
Done